File: Python/Euler013.py
#!/usr/bin/env python3
from __future__ import print_function


def main():
    digits = 10
    infile = 'input/Euler013.txt'

    nums = parse_file(infile)
    numsum = sum(nums)

    print(str(numsum)[:digits])


def parse_file(path):
    with open(path) as f:
        return [int(num) for num in f.readlines()]


if __name__ == '__main__':
    main()

File: Ruby/Euler013.rb
#!/usr/bin/env ruby

def main
  path = 'input/Euler013.txt'
  digits = 10

  nums = parse_file(path)

  sum = nums.sum

  puts sum.to_s.slice(0, 10)
end

def parse_file(path)
  File.open(path) do |f|
    f.read.split("\n")
        .filter { |line| line }
        .map { |line| line.to_i }
  end
end

if __FILE__ == $0
  main
end
